pkg/segment/reader/segread/segreader/segreader.go

```diff
@@ -226,6 +312,7 @@ func (sfr *SegmentFileReader) loadBlockUsingBuffer(blockNum uint16) (bool, error
 	if cOffAndLen.Length == 0 {
 		return false, fmt.Errorf("SegmentFileReader.loadBlockUsingBuffer: offset was 0, colName: %v, cnameIdx: %v, fname: %v", sfr.ColName, cnameIdx, sfr.fileName)
 	}
+	sfr.currRawBlockBuffer = GetBufFromPool(int64(cOffAndLen.Length))
```

Comment:
the `loadBlockUsing..` can be called multiple times. so you should fist check if `currRawBlockBuffer` is not nil, then check the size of it and see if it can fit the cOffAndLen.Lnegth, if not then return that buffer and get a new one.
otherwise we would leak